system variable: add `tidb_restricted_read_only` (#7670)
pingcap/docs: system-variables.md

    - `RESTRICTED_VARIABLES_ADMIN`: The ability to see and set sensitive variables in `SHOW [GLOBAL] VARIABLES` and `SET`.
    - `RESTRICTED_USER_ADMIN`: The ability to prevent other users from making changes or dropping a user account.

> ### tidb_restricted_read_only <span class="version-mark">New in v5.2.0</span>

Reviewer: 建议按字母顺序放在下方

- Scope: GLOBAL
- Default value: `0`

> - Optional values: `0`, `1`

Reviewer: ```suggestion
- Value optipns: `0`, `1`
```

> - This variable controls the read-only status of the entire cluster. If the variable is enabled (the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only execute the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT`, `UPDATE`, TiDB rejects to execute those statements in the read-only mode.

Author: ```suggestion
- This variable controls the read-only status of the entire cluster. If the variable is enabled (the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT` and `UPDATE`, TiDB rejects to execute those statements in the read-only mode.
```

> - This variable controls the read-only status of the entire cluster. If the variable is enabled (which means the value is `1`), all TiDB servers in the entire cluster turn on the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, `SHOW`. For others, such as `INSERT` and `UPDATE`, TiDB rejects to execute those statements in the read-only mode.

Reviewer: "reject to do" is incorrect.
```suggestion
- This variable controls the read-only status of the entire cluster. If the variable is enabled (which means that the value is `1`), all TiDB servers in the entire cluster are in the read-only mode. In this case, TiDB only executes the statements that do not modify data, such as `SELECT`, `USE`, and `SHOW`. For other statements such as `INSERT` and `UPDATE`, TiDB rejects executing those statements in the read-only mode.
```

> - If you enabled the read-only mode by this variable, this only ensures that the entire cluster finally goes into the read-only status. Suppose that you have changed the value of this variable, but the changed status is not updated to other TiDB servers. In this situation, the un-updated TiDB is still **not** in the read-only mode.

Reviewer: ```suggestion
- Enabling the read-only mode using this variable only ensures that the entire cluster will finally enter the read-only status. If you have changed the value of this variable in a TiDB cluster but the change has not yet propagated to other TiDB servers, the un-updated TiDB servers are still **not** in the read-only mode.
```

> - The read-only mode enabled by this variable only ensures that the entire cluster finally goes into the read-only status. If you have changed the value of this variable, but the changed status is not updated to other TiDB servers, the un-updated TiDB is still **not** in the read-only mode.
> - During enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements to be executed.

Author: ```suggestion
- If you enabled the read-only mode by this variable, this only ensures that the entire cluster finally goes into the read-only status. Suppose that you have changed the value of this variable, but the changed status is not updated to other TiDB servers. In this situation, the un-updated TiDB is still **not** in the read-only mode.
- While enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements **to be** executed.
```

> - While enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements **to be** executed.

Reviewer: ```suggestion
- While this variable is enabled, the SQL statements being executed are not affected. TiDB only performs the read-only check for the SQL statements **to be** executed.
```

> - When enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements that are to be executed.
> - When enabling this variable, the uncommitted transactions can have the following results:

Author: ```suggestion
- During enabling this variable, the executing SQL statements are not affected. TiDB only checks the read-only status for the SQL statements to be executed.
- During enabling this variable, the uncommitted transactions can have the following results:
```

> - While enabling this variable, the uncommitted transactions have the following results:

Reviewer: ```suggestion
- When this variable is enabled, TiDB handles the uncommitted transactions in the following ways:
```

>     - For uncommitted read-only transactions, the transactions can be committed normally.

Reviewer: ```suggestion
    - For uncommitted read-only transactions, you can commit the transactions normally.
```

>     - For uncommitted transactions that are not read-only transactions, SQL statements executing write operations in these transactions are rejected.

Reviewer: ```suggestion
    - For uncommitted transactions that are not read-only, SQL statements that perform write operations in these transactions are rejected.
```

>     - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in that transaction are rejected.
>     - If the uncommitted read-only transaction has modified data, its commitment is also rejected.

Author: ```suggestion
    - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in the transaction are rejected.
    - If the uncommitted read-only transaction has modified data, the commitment of the transaction is rejected.
```

> - During enabling this variable, the uncommitted transactions can have the following results:
>     - If there is an uncommitted read-only transaction, the transaction can be committed normally.
>     - If the uncommitted transaction is not a read-only transaction, the SQL statements executing write operations in the transaction are rejected.
>     - If the uncommitted read-only transaction has modified data, the commitment of the transaction is rejected.

Author: ```suggestion
- While enabling this variable, the uncommitted transactions have the following results:
    - For uncommitted read-only transactions, the transactions can be committed normally.
    - For uncommitted transactions that are not read-only transactions, SQL statements executing write operations in these transactions are rejected.
    - For uncommitted read-only transactions with modified data, commitments of these transactions are rejected.
```

>     - For uncommitted read-only transactions with modified data, commitments of these transactions are rejected.

Reviewer: ```suggestion
    - For uncommitted read-only transactions with modified data, the commit of these transactions is rejected.
```

> - After the cluster enables read-only mode, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.

Author: ```suggestion
- After the read-only mode is enabled, unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data.
```

> - After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.

Reviewer: ```suggestion
- After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privilege.
```

> - The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. If you enabled [Security Enhanced Mode](#tidb_enable_enhanced_security), only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.

Author: ```suggestion
- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. If the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```

> - After the read-only mode is enabled, unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data.
> - The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.

Author: ```suggestion
- After the read-only mode is enabled, all users (including the users with the `SUPER` privilege) cannot execute the SQL statements that might write data unless the user is explicitly granted the `RESTRICTED_REPLICA_WRITER_ADMIN` privileges.
- The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```

> - The users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [Security Enhanced Mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.

Reviewer: ```suggestion
- Users with `RESTRICTED_VARIABLES_ADMIN` or `SUPER` privileges can modify this variable. However, if the [security enhanced mode](#tidb_enable_enhanced_security) is enabled, only the users with the `RESTRICTED_VARIABLES_ADMIN` privilege can modify this variable.
```
